### PR TITLE
Add k8s metadata in state_cronjob metricset

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds metadata enrichment for state_cronjob metricset.

## Why is it important?
So as to be aligned with other metricsets.

## Related issues

- Closes https://github.com/elastic/beats/issues/29189

## Logs

Sample event
```
{
  "@timestamp": "2021-12-22T08:12:32.510Z",
  "@metadata": {
    "beat": "metricbeat",
    "type": "_doc",
    "version": "8.1.0"
  },
  "orchestrator": {
    "cluster": {
      "url": "kind-control-plane:6443",
      "name": "kind"
    }
  },
  "ecs": {
    "version": "8.0.0"
  },
  "host": {
    "name": "kind-control-plane"
  },
  "agent": {
    "name": "kind-control-plane",
    "type": "metricbeat",
    "version": "8.1.0",
    "ephemeral_id": "5b73917e-a756-4a78-9a26-db5fd108c1c5",
    "id": "8d46ce8b-bcd6-41f4-baa4-c2af42952625"
  },
  "service": {
    "address": "http://kube-state-metrics:8080/metrics",
    "type": "kubernetes"
  },
  "event": {
    "dataset": "kubernetes.cronjob",
    "module": "kubernetes",
    "duration": 315608278
  },
  "kubernetes": {
    "cronjob": {
      "is_suspended": false,
      "next_schedule": {
        "sec": 1.64016078e+09
      },
      "last_schedule": {
        "sec": 1.64016072e+09
      },
      "created": {
        "sec": 1.640159561e+09
      },
      "active": {
        "count": 0
      },
      "name": "hello"
    },
    "labels": {
      "foo": "bar"
    },
    "namespace": "default"
  },
  "metricset": {
    "period": 10000,
    "name": "state_cronjob"
  }
}
```

### CODE REVIEW

File: metricbeat/module/kubernetes/state_cronjob/state_cronjob_integration_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 )
 
 func TestFetchMetricset(t *testing.T) {
-	config := test.GetKubeStateMetricsConfig(t, "state_cronjob")
+	config := test.GetKubeStateMetricsConfigWithMetaDisabled(t, "state_cronjob")
```
Comment (tetianakravchenko) [inline metricbeat/module/kubernetes/state_cronjob/state_cronjob_integration_test.go:33]: why for cronjob only `add_metadata` must be disabled? as I see for other k8s resources used default value - `true`

**Review by tetianakravchenko — APPROVED** [review]
LGTM! I've left few questions

File: libbeat/common/kubernetes/metadata/resource.go [hunk 2]
```diff
@@ -118,7 +118,9 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
 			case "Deployment",
 				"ReplicaSet",
 				"StatefulSet",
-				"DaemonSet":
+				"DaemonSet",
+				"Job",
```
Comment (tetianakravchenko) [inline libbeat/common/kubernetes/metadata/resource.go:122]: with this change we also can close this PR: https://github.com/elastic/beats/pull/28954

Comment (ChrsMark) [reply libbeat/common/kubernetes/metadata/resource.go:122]: @tetianakravchenko my PR is a little bit tricky cause has some tricky parts around k8s api versions. I think we can move on with the community PR and I can rebase mine on top of it. wdyt?

File: metricbeat/module/kubernetes/state_cronjob/state_cronjob_integration_test.go (hunk 1 above)
Comment (ChrsMark) [reply metricbeat/module/kubernetes/state_cronjob/state_cronjob_integration_test.go:33]: Cause in version before v1.21 the the current stable api is not available and hence we cannot support metadata in tests for v1.20

File: libbeat/common/kubernetes/metadata/pod.go [hunk 3]
```diff
@@ -88,6 +88,7 @@ func (p *pod) GenerateK8s(obj kubernetes.Resource, opts ...FieldOptions) common.
 	out := p.resource.GenerateK8s("pod", obj, opts...)
 
 	// check if Pod is handled by a ReplicaSet which is controlled by a Deployment
+	// TODO: same happens with CronJob vs Job. The hierarcy there is CronJob->Job->Pod
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply libbeat/common/kubernetes/metadata/pod.go:91]: https://github.com/elastic/beats/issues/29771

**Comment by ChrsMark** [comment]
@MichaelKatsoulis @tetianakravchenko I will need a final review on this. I changed a little bit the behaviour and the `add_metadata` setting is disabled by default in `state_cronjob` metricset until we only support stable versions of the corresponding API.

**Review by tetianakravchenko — APPROVED** [review]
LGTM!

**Pull request merged**